databricks/koalas/series.py

```diff
@@ -4383,6 +4383,12 @@ def __len__(self):
         return len(self.to_dataframe())
 
     def __getitem__(self, key):
+        if isinstance(key, Series):
```

Comment:
@harupy, can we also check if the given Series's type is `bool` to prevent such cases:

```python
>>> pd.DataFrame({'a': [1,2,3]})[pd.Series(['a'])]
   a
0  1
1  2
2  3
```

Reply:
will fix it